music21/metadata/primitives.py

```diff
@@ -905,6 +1010,59 @@ def __init__(self, *args, **keywords):
     def _reprInternal(self):
         return f'{self.role}:{self.name}'
 
+    def __str__(self):
+        if not self.name:
+            return ''
+        return self.name
+
+    def __eq__(self, other) -> bool:
+        '''
+        >>> c1 = metadata.Contributor(role='composer', name='The Composer',
+        ...         birth='1923', death='2013')
+        >>> c2 = metadata.Contributor(role='composer', name='The Composer',
+        ...         birth='1923', death='2013')
```

Comment:
indentation again.  People need to see what is inside the parentheses and what's not.

Reply:
Can you occasionally explain why certain things return False on equality.  "Roles must be the same"  "Birth and death years must match"

Reply:
Done.